src/Grid/topology.jl

```diff
@@ -0,0 +1,341 @@
+
+############
+# Topology #
+############
+
+"""
+    getneighborhood(topology, grid::AbstractGrid, cellidx::CellIndex, include_self=false)
+    getneighborhood(topology, grid::AbstractGrid, faceidx::FaceIndex, include_self=false)
+    getneighborhood(topology, grid::AbstractGrid, vertexidx::VertexIndex, include_self=false)
+    getneighborhood(topology, grid::AbstractGrid, edgeidx::EdgeIndex, include_self=false)
+
+Returns all connected entities of the same type as defined by the respective topology. If `include_self` is true,
+the given entity is included in the returned list as well.
+"""
+getneighborhood
+
+struct EntityNeighborhood{T<:Union{BoundaryIndex,CellIndex}}
+    neighbor_info::Vector{T}
+end
+
+EntityNeighborhood(info::T) where T <: BoundaryIndex = EntityNeighborhood([info])
+Base.zero(::Type{EntityNeighborhood{T}}) where T = EntityNeighborhood(T[])
```

Comment:
I guess most of the stuff until 34 can be removed, since it was only needed for sparse matrices?

Reply:
I am still not super sure if we can eliminate this struct completely (and if this gives us more performance).

Reply:
at least for vertex connections you cannot a priori how much other vertices are connected in the general case, can you?

Reply:
I think this is true for any connectivity. I was more thinking about whether we really need EntityNeighborhood (which is basically a type wrapper around vector) or if we can use directly use a vector.

Reply:
let's delete all the functions at least that were specified in order to make `SparseMatrixCSC` usage work